QUARKUS-1079.md

```diff
@@ -0,0 +1,74 @@
+# QUARKUS-1079 - Hibernate Reactive as Tech Preview
+
+JIRA link: https://issues.redhat.com/browse/QUARKUS-1079
+
+For native mode similar test coverage as for JVM mode needs to be ensured.
+
+## Scope of the testing
+This feature added previews of two extensions: 
+ - Hibernate Reactive
+ - Hibernate Reactive with Panache
+
+Following test coverage should be implemented for Hibernate Reactive:
+- Usage of application.properties file for passing reactive-hibernate settings [examples]
+- Usage of import.sql [examples]
+- Smoke test(getting a working Mutiny session) [examples]
+- Retrieving Multi result from the DB [examples]
+- Retrieving Uni result(require a transformation) from the DB
+- Usage of DB-generated ids and ids generated with Java code(via ReactiveIdentifierGenerator) [examples]
+- Validation of values [examples, doesn't work as expected]
+- Attribute converters
+- Usage of CompletionStages alongside a Mutiny API
+- Usage of session opened through openSession() method as well as withSession() [examples]
+- Usage of transaction started through openSession() method as well as withSession()
+- Usage of session and transaction opened with the same method
+- Session methods: find[examples], persist[examples], remove, merge, flush[examples]
+- Queries: regular[examples], native, named, JPA criteria
+- Query methods: setParameter[examples], setMaxResults, getFirstResult, getSingleResult[examples], getResultList[examples], executeUpdate
+- Eager[examples] and lazy fetches
+- Field-level lazy fetching(Hibernate team "doesn't encourage" using this feature, though)
+- Stateless sessions and their methods
+ 
+Following test coverage should be implemented for Hibernate Reactive with Panache:
+- Active record pattern
+- Repository pattern
+- list and stream methods
+- find method
+- sorting 
+- Named queries
+- DTO projection
+- Transactions
+- Id generation
+
+Ideally, all of aforementioned tests should run for PostgreSQL[examples], MySQL and DB2 connections.
+
+### Impact on testsuites and testing automation:
+ - Ensure this coverage works both in JVM and NATIVE mode
+ - Some smoke tests were created in "examples" section of quarkus-test-framework, more advanced should be put into quarkus-test-suite
+
+### Impact on resources:
+ - More services will be running on OpenShift cluster, impact on CPU/memory should be minimal
+ - Bare metal tests will deploy docker containers with DBs, but they are mostly lightweight
```

Comment:
DB2 for example takes minutes to start and is heavy on memory

Reply:
Do we spin db2 on Openshift directly? In Vertx we were using dballocator for this purpose.

Reply:
We do not use dballocator to avoid hard dependency on this service, DBs are started in container to have better portability and easier local investigation if case fails.

Reply:
@rsvoboda I have two questions:
1) Which databases should we cover? Is there is a list or something?
2) Are any of them comparable to db2 in terms of speed and resources used? I checked Postgres, Maria, My  and MS SQL, and they seem to be relatively fast. Brief search for resource consumption showed it to be below 2 GBs, which is half of what is required for Docker, so it should not be a problem.

Reply:
Details are in the umbrella epic in JIRA - https://issues.redhat.com/browse/QUARKUS-962
Thomas wrote:
 -  PostgreSQL and MySQL is required
 -  DB2, MSSQL and Oracle as tech preview is important but not required

MS SQL reactive client was added in https://github.com/quarkusio/quarkus/releases/tag/2.1.0.CR1
From my side we need to cover also TP databases as they will be promoted to fully supported at some time.

DB2 is probably the worse from CPU/RAM perspective.

Reply:
Added info about DB2